tests/sample.js

```diff
@@ -0,0 +1,199 @@
+  var Bridge = function(param) {
+    console.log(param.type);
+    this.bridge_type = param.type;
+    this.id = "1";
+    this.addChannel= function(input, cb) {
+      console.log("TREX");
+      pastStasis = 1;
+      var channel = channels.filter(function(testChan) {
+        return testChan.id === '1';
+      })[0];
+      bridge_channels.push(channel);
+      cb(null);
+      console.log(channel);
+      this.emit('ChannelEnteredBridge', {channel: {id: channel.id}});
+    };
+  };
+  util.inherits(Bridge, Emitter);
+  mockBridge = new Bridge(param);
+  return mockBridge;
+};
+
+/**
+ * Returns a mock channel that has an id, and can use originate to add a channel
+ * to the array of total channels. Emits a StasisStart event when originating.
+ */
+
+var getMockChannel = function() {
+  var Channel = function() {
+    this.id = '1';
+    this.originate = function(input, cb) {
+      console.log(input);
+      var self = this;
+      setTimeout(function() {
+        if(valid == 1) {
+          console.log(self);
+          self.emit('StasisStart', self, self);
+        }
+      }, 20);
+    };
+  };
+  util.inherits(Channel, Emitter);
+  mockChannel = new Channel();
+  channels.push(mockChannel);
+  return mockChannel;
+};
+
+describe('SLA Bridge and Channel Tester', function() {
+
+  afterEach(function(done) {
+    channels = [];
+    bridge_channels = [];
+    bridge_created = 0;
+    done();
+    });
+
+  it('should create a bridge when there isn\'t one', function(done) {
+    var channel = getMockChannel();
+    var client = getMockClient();
+    var sla = require('../lib/sla.js')(client, end);
+
+    bridgeChecking();
+    function bridgeChecking() {
+      setTimeout(function() {
+        if(channels.length != 0 && bridge_created == 1) {
+          done();
+        } else {
+          bridgeChecking();
+        }
+      }, asyncDelay);
+    } 
+  });
+
+  it('should use a preexisting bridge if there is one', function(done) {
+    var channel = getMockChannel();
+    var client = getMockClient();
+    var sla = require('../lib/sla.js')(client, end);
+
+    bridgeChecking();
+    function bridgeChecking() {
+      setTimeout(function() {
+        if(channels.length != 0 && bridge_created == 0) {
+          done();
+        } else {
+          bridgeChecking();
+        }
+      }, asyncDelay);
+    } 
+  });
+
+  it('should use a "valid" endpoint and give off a StasisStart event ' +
+     'to proceed to the next section', function(done) {
+    var channel = getMockChannel();
+    var client = getMockClient();
+    var sla = require('../lib/sla.js')(client, end);
+
+    validChecking();
+    function validChecking() {
+      setTimeout(function() {
+        if(pastStasis == 1 && bridge_channels.length != 0) {
+          pastStasis = 0;
+          done();
+        } else {
+          validChecking();
+        }
+      }, asyncDelay);
+    } 
+  });
+
+  it('should use an "invalid" endpoint and not give off a StasisStart event',
```

Comment:
This test is no different from the previous one, it's only passing because the check occurs before the call updates the variables you are checking. Until sla.js is made to expect parameters to drive what endpoint is called you may not be able to test this.